config.c

```diff
@@ -0,0 +1,294 @@
+/**
+ * Config Reader and Parser
+ * This program will read the config for lightwm in the appdata directory
+ * 
+ * Demetry Romanowski
+ * demetryromanowski@gmail.com
+ **/ 
+#include "config.h" 
+
+#include <Windows.h> 
+#include <Shlobj.h> 
+#include <wchar.h> 
+#include <stdio.h> 
+#include <strsafe.h> 
+#include <shlwapi.h> 
+#include <assert.h> 
+
+#include "error.h" 
+#include "resource.h"
+
+#include "debug.h"
+
+#define BUFF_SIZE 65536
+
+/**
+ * Config reader global vars
+ **/
+PWSTR szConfigFilePath[MAX_PATH]; 
+char* defaultConfigData = NULL; 
+
+//Should probably create a meta structure that holds the total count for now just another global variable
+ConfigItems configItems; 
+
+/**
+ * Private prototypes here
+**/
+BOOL CreateDefaultConfigFile(HINSTANCE);
+BOOL LoadDefaultConfigResourceData(HINSTANCE);
+BOOL WriteDefaultConfigDataToFile(); 
+void trim(char* str); 
+void removeControlChars(char* str); 
+size_t GetLineCount(FILE* file);
+ 
+DWORD ReadConfigFile() 
+{ 
+	//Try to open the config file based on the path
+	FILE* configFileHandle = _wfopen(*szConfigFilePath, L"r"); 
+	
+	if(configFileHandle == NULL) 
+	{
+		SetLastError(ERROR_INVALID_HANDLE); 
+		reportWin32Error(L"Config file could not be opened"); 
```

Comment:
Since this is not a Win32 Error but a C library function that failed, I would add another function similar to reportWin32Error, but for other kinds of errors